recompute_test_point fails after pickling

## Recompute_test_point fails after pickling


**Please provide a minimal, self-contained, and reproducible example.**
```python
def test_pickling():
    import cloudpickle
    with pm.Model() as model:
        pm.Normal("one")
    model = cloudpickle.loads(cloudpickle.dumps(model))
    model.recompute_initial_point()
```

**Please provide the full traceback.**

<details><summary>Complete error traceback</summary>

```python
>       model.recompute_initial_point()

pymc/tests/test_initial_point.py:74:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pymc/model.py:945: in recompute_initial_point
    fn = make_initial_point_fn(model=self, return_transformed=True)
pymc/initial_point.py:185: in make_initial_point_fn
    graph.replace_all(zip(rng_nodes, new_rng_nodes), import_missing=True)
../../.pyenv/versions/3.8.7/envs/pymc3/lib/python3.8/site-packages/aesara/graph/fg.py:554: in replace_all
    self.replace(var, new_var, **kwargs)
../../.pyenv/versions/3.8.7/envs/pymc3/lib/python3.8/site-packages/aesara/graph/fg.py:515: in replace
    new_var = var.type.filter_variable(new_var, allow_convert=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = RandomStateType, other = RandomStateSharedVariable(<RandomState(PCG64) at 0x7FEE7A091240>), allow_convert = True

    def filter_variable(
        self, other: Union[Variable, D], allow_convert: bool = True
    ) -> Variable:
        r"""Convert a symbolic variable into this `Type`, if compatible.

        For the moment, the only `Type`\s compatible with one another are
        `TensorType` and `GpuArrayType`, provided they have the same number of
        dimensions, same broadcasting pattern, and same dtype.

        If `Type`\s are not compatible, a ``TypeError`` should be raised.

        """
        if not isinstance(other, Variable):
            # The value is not a Variable: we cast it into
            # a Constant of the appropriate Type.
            other = self.Constant(type=self, data=other)

        if other.type != self and allow_convert:
            other2 = self.convert_variable(other)
            if other2 is not None:
                return other2

        if other.type != self:
>           raise TypeError(
                "Cannot convert Type %(othertype)s "
                "(of Variable %(other)s) into Type %(self)s. "
                "You can try to manually convert %(other)s into a %(self)s."
                % dict(othertype=other.type, other=other, self=self)
            )
E           TypeError: Cannot convert Type RandomStateType (of Variable RandomStateSharedVariable(<RandomState(PCG64) at 0x7FEE7A091240>)) into Type RandomStateType. You can try to manually co
nvert RandomStateSharedVariable(<RandomState(PCG64) at 0x7FEE7A091240>) into a RandomStateType.
```

</details>

Related to #4582 


## Versions and main components

* PyMC/PyMC3 Version: master
* Aesara/Theano Version: master
* Python Version: 3.9
* Operating system: Linux
* How did you install PyMC/PyMC3: development

--- a/conda-envs/environment-dev-py37.yml
+++ b/conda-envs/environment-dev-py37.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-dev-py38.yml
+++ b/conda-envs/environment-dev-py38.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-dev-py39.yml
+++ b/conda-envs/environment-dev-py39.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-test-py37.yml
+++ b/conda-envs/environment-test-py37.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-test-py38.yml
+++ b/conda-envs/environment-test-py38.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-test-py39.yml
+++ b/conda-envs/environment-test-py39.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.4
 - cachetools
 - cloudpickle

--- a/conda-envs/windows-environment-dev-py38.yml
+++ b/conda-envs/windows-environment-dev-py38.yml
@@ -5,7 +5,7 @@ channels:
 dependencies:
  # base dependencies (see install guide for Windows)
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/windows-environment-test-py38.yml
+++ b/conda-envs/windows-environment-test-py38.yml
@@ -5,7 +5,7 @@ channels:
 dependencies:
  # base dependencies (see install guide for Windows)
 - aeppl>=0.0.13
-- aesara>=2.2.2
+- aesara>=2.2.6
 - arviz>=0.11.2
 - cachetools
 - cloudpickle

--- a/pymc/tests/test_initial_point.py
+++ b/pymc/tests/test_initial_point.py
@@ -13,6 +13,7 @@
 #   limitations under the License.
 import aesara
 import aesara.tensor as at
+import cloudpickle
 import numpy as np
 import pytest
 
@@ -218,3 +219,12 @@ class TestMoment:
             assert not hasattr(rv.tag, "test_value")
             assert tuple(get_moment(rv).shape.eval()) == (4, 3)
         pass
+
+
+def test_pickling_issue_5090():
+    with pm.Model() as model:
+        pm.Normal("x", initval="prior")
+    ip_before = model.recompute_initial_point(seed=5090)
+    model = cloudpickle.loads(cloudpickle.dumps(model))
+    ip_after = model.recompute_initial_point(seed=5090)
+    assert ip_before["x"] == ip_after["x"]

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -2,7 +2,7 @@
 # See that file for comments about the need/usage of each dependency.
 
 aeppl>=0.0.13
-aesara>=2.2.2
+aesara>=2.2.6
 arviz>=0.11.4
 cachetools>=4.2.1
 cloudpickle

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 aeppl>=0.0.13
-aesara>=2.2.2
+aesara>=2.2.6
 arviz>=0.11.4
 cachetools>=4.2.1
 cloudpickle

